webapp/src/pages/UserStatsPage/UserStatsPage.tsx

```diff
@@ -17,44 +16,107 @@ import 'chartjs-adapter-date-fns';
+  const { data: user, isLoading: isUserLoading } = useData(getUserByUserName);
+  const getLevelHistory = useCallback(async (): Promise<
+    FiveArray<UserLevelWithTimestamp[]>
+  > => {
+    try {
+      if (!user) {
+        return Promise.reject('Unresolved');
+      }
+      let pclassic: proArray = [
+        await gameApi.gameControllerGetUserLevelHistory({
+          username: username!,
+          mode: GameControllerGetUserLevelHistoryModeEnum.Classic,
+        }),
+        GameControllerGetUserLevelHistoryModeEnum.Classic,
+      ];
+      let pmystery: proArray = [
+        await gameApi.gameControllerGetUserLevelHistory({
+          username: username!,
+          mode: GameControllerGetUserLevelHistoryModeEnum.MysteryZone,
+        }),
+        GameControllerGetUserLevelHistoryModeEnum.MysteryZone,
+      ];
+      let pshort: proArray = [
+        await gameApi.gameControllerGetUserLevelHistory({
+          username: username!,
+          mode: GameControllerGetUserLevelHistoryModeEnum.ShortPaddle,
+        }),
+        GameControllerGetUserLevelHistoryModeEnum.ShortPaddle,
+      ];
+      let ptrain: proArray = [
+        await gameApi.gameControllerGetUserLevelHistory({
+          username: username!,
+          mode: GameControllerGetUserLevelHistoryModeEnum.Training,
+        }),
+        GameControllerGetUserLevelHistoryModeEnum.Training,
+      ];
+      let punknown: proArray = [
+        await gameApi.gameControllerGetUserLevelHistory({
+          username: username!,
+          mode: GameControllerGetUserLevelHistoryModeEnum.Unknown,
+        }),
+        GameControllerGetUserLevelHistoryModeEnum.Unknown,
+      ];
+      [pclassic, pmystery, pshort, ptrain, punknown].map((levels) => {
+        return levels[0].unshift({
+          username: user.username,
+          timestamp: user.createdAt,
+          gameMode: levels[1],
+          level: 1,
+          gameId: '', // maybe we should allow null in gameId here,
+        });
+      });
+      return [pclassic[0], pmystery[0], pshort[0], ptrain[0], punknown[0]];
+    } catch (e) {
+      return Promise.reject('Unresolved');
+    }
+  }, [user, username]);
+  const { data: levels, isLoading: areLevelsLoading } = useData<
+    FiveArray<UserLevelWithTimestamp[]>
+  >(
+    getLevelHistory,
+    useCallback(() => {}, []),
+  );
   const getStats = useCallback(
     () =>
-      gameApi.gameControllerGetUserStats({
-        username: username!,
-        mode: GameControllerGetUserStatsModeEnum.Classic,
-      }),
+      Promise.all(
+        [
+          GameControllerGetUserStatsModeEnum.Classic,
+          GameControllerGetUserStatsModeEnum.ShortPaddle,
+          GameControllerGetUserStatsModeEnum.MysteryZone,
+          GameControllerGetUserStatsModeEnum.Training,
+          GameControllerGetUserStatsModeEnum.Unknown,
+        ].map((mode) => {
+          return gameApi.gameControllerGetUserStats({
+            username: username!,
+            mode,
```

Comment:
mode is optional. If I don't pass it, I get all of them, and then I don't need the reducer.

Reply:
will do.

Reply:
done